Automattic/jetpack
# Contact Form: Revised the email validation regex to include length limit

Fixes #3734

#### Changes proposed in this Pull Request:

* Revised the email validation regex to include length limit, in order to match the latest practical implementation of RFC 5322 in [regular-expressions.info/email.html](https://www.regular-expressions.info/email.html).

* Regex used:

PHP version: the very bottom one on [regular-expressions.info/email.html](https://www.regular-expressions.info/email.html). The one after the sentence "...modern regex flavors aren't truly regular, so we can add length limit checks using lookahead like we did before:" Take note, the redundant pink-colored white spaces in the middle of the regex must be removed.

```
\A(?=[a-z0-9@.!#$%&'*+/=?^_`{|}~-]{6,254}\z)(?=[a-z0-9.!#$%&'*+/=?^_`{|}~-]{1,64}@)[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:(?=[a-z0-9-]{1,63}\.)[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+(?=[a-z0-9-]{1,63}\z)[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\z
```
JavaScript version:

```
^(?=[a-z0-9@.!#$%&'*+/=?^_\`{|}~-]{6,254}$)(?=[a-z0-9.!#$%&'*+/=?^_\`{|}~-]{1,64}@)[a-z0-9!#$%&'*+/=?^_\`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_\`{|}~-]+)*@(?:(?=[a-z0-9-]{1,63}\.)[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+(?=[a-z0-9-]{1,63}$)[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$
```

#### Testing instructions:



1.) Checkout the code in this PR on a test WP site.
2.) Use the Contact Form and enter some test data in each field.
3.) In the email field, test different email addresses that are expected to pass and fail.
4.) Use an email you've access to, submit the form and make sure you get the email.
5.) Revert the code back to its previous state.


#### Proposed changelog entry for your changes:
* Contact Form: Revised the email validation regex to include length limit.
